### PR TITLE
[setup] scaffold for application preview

### LINKED ISSUE #171: display info on app card pop-ups
yay splendid job last week!! 🍄

This week, you’ll be working on displaying the information for each app card pop-up. It will mostly be unstyled (and minimal functionality) until Jesslyn’s hi-fis are ready. For now just route to a pop-up for each of these when you click on an application card’s button. Here’s the breakdown by status:

- incomplete → route to the application (the exception, so no pop-up just confirm this still works)
- pending → display the application info + message (“Admin is currently reviewing your application.”)
    - app info = same as that of the main application review page along with the ranked adoptees
- pending confirmation → display matched adoptee info + set up confirmation flow 
    - use the new `matched_adoptee` col from adopter application table 
    - no functionality for now! just frontend, have a question at the bottom of the pop-up for “Do you confirm…?” with yes/no buttons. no option should render text input for “Why?” 
- active → display matched adoptee info + set up end flow
    - display same matched adoptee info as pending confirmation
    - again no functionality, but at bottom have “Select a reason for ending your application.” and a dropdown with placeholders (will update style later)
- reapply → display application OR match info + corresponding message
    - if `matched_adoptee` is null, then display application info (same as pending) + message “Admin found an issue with your application, create a new one.”
    - if `matched_adoptee` not null, then display basic matched adoptee info (same as pending confirmation) + message “You didn’t respond in time, create a new application.”
- rejected → display application info (same as pending) + message (“You are blocked from creating future applications. Contact admin to fix this.”)
- ended → display basic adoptee info + reason/date for ending
    - create a new col for `time_ended`, and rename `return_explanation` to `ended_reason` for the ending reason/date

Some of the messages and a lot of the design will change over time, but this will help with setting up the functionality and hi-fis later. You can use “app card lo-fi/drafts (inaya)” from hi-fis/desktop in the figma as a rough visual.

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Implemented scaffold for application preview for different statuses
- Added relevant formatter functions to format various application values to a display-appropriate string
- Removed main application question for return reason (deprecated)
- Refactored dialog controls on dashboard (now mutually exclusive with email confirmation dialog)

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

General

https://github.com/user-attachments/assets/c91c3c95-48a0-4ce4-b142-8ddbc31468c2

Status: `PENDING`
image

Status: `PENDING_CONFIRMATION`

option: yes
image

option: no
image

Status: `ACCEPTED`

https://github.com/user-attachments/assets/888d9f19-cb75-4c1b-98b2-90ce50b6339c

Status: `REJECTED`
image

Status: `REAPPLY`

no match
image

matched
image

Status: `ENDED`

`ended_reason` and `time_ended` are both null
image

with `time_ended` only
image

with `ended_reason` only
image

with `ended_reason` and `time_ended`
image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Verify:
- Clicking on application card on dashboard opens a pop up
- Clicking away closes it
- Pressing `ESCAPE` closes the pop up (can disable)
- After closing pop-up, pressing the back button (mouse or browser) navigates back
    - Verify any other such browser navigation maneuvers work as intended
- Test different application statuses (modify on Supabase)
    - Vary other values, like `matched_adoptee`, `ended_reason`, `time_submitted`, `time_ended`
- Invalid inputs to controls (e.g. rejecting a confirmation) should not trigger a submission
    - If a submission was triggered, results will be printed to the browser console

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Finalize text message and design when hi-fis are completed
- Replace hard-coded ended reasons with actual reasons
- Update Supabase when user forms are submitted

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

- [Radix Primitive: Collapsible](https://www.radix-ui.com/primitives/docs/components/collapsible#animating-content-size)
- [Tailwind: Animation](https://tailwindcss.com/docs/animation#using-a-custom-value)
- [Tailwind: Data Attributes](https://tailwindcss.com/docs/hover-focus-and-other-states#data-attributes)

CC: @InayaY3

### CODE REVIEW

File: components/home/application-preview/StatusPill.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { cva } from 'class-variance-authority';
+import { formatAppStatus } from '@/lib/formatters';
+import { AdopterApplication } from '@/types/schema';
+
+const statusPillStyles = cva('rounded-xl px-3 py-0.5 shadow-sm', {
+  variants: {
+    status: {
+      PENDING: 'bg-[#FFF8D0] text-[#D39C49]',
+      PENDING_CONFIRMATION: 'bg-[#FFF8D0] text-[#D39C49]',
+      ACCEPTED: 'bg-[#DEF4DF] text-[#3E6430]',
```
Comment (InayaY3) [inline components/home/application-preview/StatusPill.tsx:10]: what is this for? shouldn't we delete this? it's not being used anywhere

File: components/MainDashboard/MainDashboardTabs.tsx [hunk 2]
```diff
@@ -36,19 +35,6 @@ export default function MainDashBoardTabs() {
...
-        .eq('user_id', user.id)
-        .maybeSingle();
-
-      if (profileData?.first_name)
-        setDisplayName(
-          [profileData.first_name, profileData.last_name]
-            .filter(Boolean)
-            .join(' '),
-        );
-
```
Comment (InayaY3) [inline components/MainDashboard/MainDashboardTabs.tsx:51]: not displaying on nav bar (logout or greeting), no reason to put adopter name on app cards either. meant to remove earlier

File: components/MainDashboard/ApplicationCard.tsx [hunk 3]
```diff
@@ -1,30 +1,43 @@
...
+      className="flex h-60 w-96 flex-col rounded-2xl border-2 border-cyan-12 bg-white"
+    >
+      <div className="flex flex-col gap-y-30">
+        <div className="flex flex-row justify-end pt-3 pr-3">
+          <ApplicationCardButton app={app} />
+        </div>
+
+        <div className="flex flex-col pb-3 pl-3">
+          <p className="font-medium text-red-12">
+            {app.time_submitted
```
Comment (InayaY3) [inline components/MainDashboard/ApplicationCard.tsx:32]: had to update some styling + rebase to match basics of dashboard PR

File: components/home/application-preview/ApplicationPreviewDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+  if (!(showPreview && previewId && appData?.time_submitted)) return null;
+
+  // replace URL to / when modal closes
+  const triggerNavigate = (open: boolean) => {
+    if (open) return;
+    router.replace(
+      historyStatuses.includes(appData.status) ? '/?tab=history' : '/',
+    );
+  };
```
Comment (InayaY3) [inline components/home/application-preview/ApplicationPreviewDialog.tsx:97]: not the best fix i assume. but for now. also the nav bar highlight glitches so gotta fix that later. noted alr but js saying. (this is all bc we have a separate page for history instead of everything being on the dashboard. shouldnt be an issue for the confirmation routing tho)

File: components/MainDashboard/MainDashboardTabs.tsx [hunk 5]
```diff
@@ -91,11 +77,7 @@ export default function MainDashBoardTabs() {
       <div className="grid grid-cols-2 gap-8">
         {!showHistory && true}
```
Comment (InayaY3) [inline components/MainDashboard/MainDashboardTabs.tsx:78]: also need to fix so that reapply is part of history. shows on main rn

File: components/home/application-preview/ApplicationPreviewDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      }
+
+      setAppData(data);
+    };
+
+    fetchData();
+  }, [previewId]);
+
+  // determine app status/step
+  const statusText = useMemo(() => {
```
Comment (InayaY3) [inline components/home/application-preview/ApplicationPreviewDialog.tsx:56]: maybe misinterp but assuming active will be added eventually. and adoptee info in gen for cases where we don't just want app info. i saw it in some cases and was able to test by manually adding an id, which worked!

**Review by InayaY3 — APPROVED** [review]
sorryyyyyy i had to rebase. i assume most changes will be made in the actual styling PR, so i may not merge that yet
